### ISSUE TITLE
Results on Fashion-MNIST is worse than results on your paper

### ISSUE THREAD

**Issue by sarsbug** [rymc/n2d#8]
I tried to run n2d on fashin-mnist dataset,but I got the result like this which is worse than results on your paper.
===========================================================================
0.51409
0.55714
0.37784
===========================================================================
I used the same parameters in the run.sh and the trained model downloaded with the code.
Can you please tell me how to acheive the ACC:0.672 and NMI:0.684 results on the dataset?

**Comment by rymc** [comment]
Hi,

Thank you for raising this issue. 

I have done a fresh clone of the repo on a new virtual environment and was able to reproduce the results (within ~1% or so the reported results).

In order to help me understand what is going on can you tell me if you followed the guide in the readme (create a new environment with conda, pip install -r requirements.txt etc.)?

**Comment by sarsbug** [comment]
> Hi,
> 
> Thank you for raising this issue.
> 
> I have done a fresh clone of the repo on a new virtual environment and was able to reproduce the results (within ~1% or so the reported results).
> 
> In order to help me understand what is going on can you tell me if you followed the guide in the readme (create a new environment with conda, pip install -r requirements.txt etc.)?

I just used my environment. Thanks for replying, I will try to create a new one with the instruction.

**Comment by sarsbug** [comment]
I tried to create the same environment as yours but the result is still worse.
fashion | UMAP on autoencoded embedding with GMM - N2D
================================================================================
0.55284
0.55883
0.40102
================================================================================

I used this script to test Fashion-MNIST:
python n2d.py fashion 0 --ae_weights=fashion-1000-ae_weights.h5  --umap_dim=10 --umap_neighbors=20 --manifold_learner=UMAP  --save_dir=fashion-n2d --umap_min_dist=0.00


And this is my conda environment:
# packages in environment at /home/zxz/anaconda3/envs/n2d:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
absl-py                   0.9.0                    pypi_0    pypi
astor                     0.8.1                    pypi_0    pypi
ca-certificates           2020.1.1                      0    defaults
certifi                   2019.11.28               py37_0    defaults
cycler                    0.10.0                   pypi_0    pypi
gast                      0.3.3                    pypi_0    pypi
grpcio                    1.27.2                   pypi_0    pypi
h5py                      2.10.0                   pypi_0    pypi
joblib                    0.14.1                   pypi_0    pypi
keras                     2.2.4                    pypi_0    pypi
keras-applications        1.0.8                    pypi_0    pypi
keras-preprocessing       1.1.0                    pypi_0    pypi
kiwisolver                1.1.0                    pypi_0    pypi
ld_impl_linux-64          2.33.1               h53a641e_7    defaults
libedit                   3.1.20181209         hc058e9b_0    defaults
libffi                    3.2.1                hd88cf55_4    defaults
libgcc-ng                 9.1.0                hdf63c60_0    defaults
libstdcxx-ng              9.1.0                hdf63c60_0    defaults
llvmlite                  0.31.0                   pypi_0    pypi
markdown                  3.2.1                    pypi_0    pypi
matplotlib                3.1.0                    pypi_0    pypi
mock                      4.0.1                    pypi_0    pypi
ncurses                   6.2                  he6710b0_0    defaults
numba                     0.48.0                   pypi_0    pypi
numpy                     1.16.4                   pypi_0    pypi
openssl                   1.1.1d               h7b6447c_4    defaults
pandas                    0.24.2                   pypi_0    pypi
pip                       20.0.2                   py37_1    defaults
protobuf                  3.11.3                   pypi_0    pypi
pyparsing                 2.4.6                    pypi_0    pypi
python                    3.7.6                h0371630_2    defaults
python-dateutil           2.8.1                    pypi_0    pypi
pytz                      2019.3                   pypi_0    pypi
pyyaml                    5.3                      pypi_0    pypi
readline                  7.0                  h7b6447c_5    defaults
scikit-learn              0.21.3                   pypi_0    pypi
scipy                     1.4.1                    pypi_0    pypi
seaborn                   0.9.0                    pypi_0    pypi
setuptools                45.2.0                   py37_0    defaults
six                       1.14.0                   pypi_0    pypi
sqlite                    3.31.1               h7b6447c_0    defaults
tensorboard               1.13.1                   pypi_0    pypi
tensorflow                1.13.1                   pypi_0    pypi
tensorflow-estimator      1.13.0                   pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
tk                        8.6.8                hbc83047_0    defaults
umap-learn                0.3.9                    pypi_0    pypi
werkzeug                  1.0.0                    pypi_0    pypi
wheel                     0.34.2                   py37_0    defaults
xz                        5.2.4                h14c3975_4    defaults
zlib                      1.2.11               h7b6447c_3    defaults

**Comment by rymc** [comment]
Thanks for the update. Can you run md5sum weights/fashion-1000-ae_weights.h5 and reply with the output?

Best wishes
Ryan

**Comment by sarsbug** [comment]
> Thanks for the update. Can you run md5sum weights/fashion-1000-ae_weights.h5 and reply with the output?
> 
> Best wishes
> Ryan

Thank for your reply. This is my md5sum result of model weight and my dataset.

md5sum weights/fashion-1000-ae_weights.h5
211d12e7f7995e9aa367a23abb2c6be4  weights/fashion-1000-ae_weights.h5

9fb629c4189551a2d022fa330f9573f3  t10k-images-idx3-ubyte.gz
bb300cfdad3c16e7a12a480ee83cd310  t10k-labels-idx1-ubyte.gz
8d4fb7e6c68d591d4c3dfef9ec88bf0d  train-images-idx3-ubyte.gz
25c81989df183df01b3e8a0aad5dffbe  train-labels-idx1-ubyte.gz

**Comment by rymc** [comment]
This is strange. I have deployed this on three different machines and achieve similar results to what I reported in the paper.

Can you zip up your n2d directory and send it to me via email (github@ryanmcconville.com)?

Best wishes
Ryan

**Comment by sarsbug** [comment]
> This is strange. I have deployed this on three different machines and achieve similar results to what I reported in the paper. Can you zip up your n2d directory and send it to me via email (github@ryanmcconville.com)? Best wishes Ryan
> […](#)
> On 2 Mar 2020, 14:04 +0000, sarsbug ***@***.***>, wrote: > Thanks for the update. Can you run md5sum weights/fashion-1000-ae_weights.h5 and reply with the output? > Best wishes > Ryan Thank for your reply. This is my md5sum result of model weight and my dataset. md5sum weights/fashion-1000-ae_weights.h5 211d12e7f7995e9aa367a23abb2c6be4 weights/fashion-1000-ae_weights.h5 9fb629c4189551a2d022fa330f9573f3 t10k-images-idx3-ubyte.gz bb300cfdad3c16e7a12a480ee83cd310 t10k-labels-idx1-ubyte.gz 8d4fb7e6c68d591d4c3dfef9ec88bf0d train-images-idx3-ubyte.gz 25c81989df183df01b3e8a0aad5dffbe train-labels-idx1-ubyte.gz — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

I have already sent you a email with my n2d directory. Thanks for helping.

**Comment by sarsbug** [comment]
I don't know why I downloaded the wrong dataset from webset. 

When I tried to download the dataset by fashion_mnist.load_data() of keras I got the good result as the paper. Apperiate for @rymc 's help.
